src/utilities/utility.ts

```diff
@@ -50,50 +51,75 @@ export function checkForErrors(execResults: IExecSyncResult[], warnIfError?: boo
+export function getLastSuccessfulRunSha(kubectl: Kubectl, namespaceName: string, annotationKey: string): string {
+    const result = kubectl.getResource('namespace', namespaceName);
+    if (!result) {
+        core.debug(`Failed to get commits from cluster.`);
+        return '';
+    }
+    else {
+        if (!!result.stderr) {
+            core.debug(`${result.stderr}`);
+            return process.env.GITHUB_SHA;
+        }
+        else if (!!result.stdout) {
+            const annotationsSet = JSON.parse(result.stdout).metadata.annotations;
+            if (!!annotationsSet && !!annotationsSet[annotationKey]) {
+                return JSON.parse(annotationsSet[annotationKey].replace(/'/g, '"')).commit;
+            }
+            else {
+                return 'NA';
+            }
+        }
+    }
+}
+
+export async function getWorkflowFilePath(githubToken: string): Promise<string> {
+    let workflowFilePath = process.env.GITHUB_WORKFLOW;
+    if (!workflowFilePath.startsWith('.github/workflows/')) {
+        const githubClient = new GitHubClient(process.env.GITHUB_REPOSITORY, githubToken);
+        const response = await githubClient.getWorkflows();
+        if (response.statusCode == StatusCodes.OK
+            && !!response.body
+            && !!response.body.total_count) {
+            if (response.body.total_count > 0) {
+                for (let workflow of response.body.workflows) {
+                    if (process.env.GITHUB_WORKFLOW === workflow.name) {
+                        workflowFilePath = workflow.path;
+                        break;
+                    }
+                }
+            }
+        }
+        else if (response.statusCode != StatusCodes.OK) {
+            core.debug(`An error occured while getting list of workflows on the repo. Statuscode: ${response.statusCode}, StatusMessage: ${response.statusMessage}`);
+        }
+    }
+    return Promise.resolve(workflowFilePath);
+}
+
+export function annotateChildPods(kubectl: Kubectl, resourceType: string, resourceName: string, annotationKeyValStr: string, allPods): IExecSyncResult[] {
     const commandExecutionResults = [];
     let owner = resourceName;
     if (resourceType.toLowerCase().indexOf('deployment') > -1) {
         owner = kubectl.getNewReplicaSet(resourceName);
     }
 
-    if (allPods && allPods.items && allPods.items.length > 0) {
+    if (!!allPods && !!allPods.items && allPods.items.length > 0) {
         allPods.items.forEach((pod) => {
             const owners = pod.metadata.ownerReferences;
-            if (owners) {
+            if (!!owners) {
                 owners.forEach(ownerRef => {
                     if (ownerRef.name === owner) {
-                        commandExecutionResults.push(kubectl.annotate('pod', pod.metadata.name, workflowAnnotations, true));
+                        commandExecutionResults.push(kubectl.annotate('pod', pod.metadata.name, [annotationKeyValStr], true));

```

Comment:
Is it a single `annotation`? Or are you converting the array into a string and passing it here? 

If it's going to be a single string, consider changing the signature of the function

Reply:
It is a single key with JSON object as value. Array is not converted into string here. Kept the signature as array to felicitate multiple annotations for future scenarios. Want me to remove it?

Reply:
Yes, the current scenario clearly doesn't warrant this.